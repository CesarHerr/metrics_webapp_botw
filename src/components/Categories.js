import { NavLink } from 'react-router-dom';
import '../styles/Categories.css';

function Categories() {
  return (
    <ul className="categoryList">
      <li className="creatures">

        <NavLink to="creatures" className="test">
          <span className="material-symbols-outlined">
            arrow_circle_right
          </span>
          <div>
            <h2>Creatures</h2>
            <h3>82</h3>
          </div>
        </NavLink>

      </li>
      <li className="equipment">

        <NavLink to="equipment">
          <span className="material-symbols-outlined">
            arrow_circle_right
          </span>
          <div>
            <h2>Equipment</h2>
            <h3>184</h3>
          </div>
        </NavLink>

      </li>
      <li className="materials">

        <NavLink to="materials">
          <span className="material-symbols-outlined">
            arrow_circle_right
          </span>
          <div>
            <h2>Materials</h2>
            <h3>80</h3>
          </div>
        </NavLink>

      </li>
      <li className="monsters">

        <NavLink to="monsters">
          <span className="material-symbols-outlined">
            arrow_circle_right
          </span>
          <div>
            <h2>Monsters</h2>
            <h3>80</h3>
          </div>
        </NavLink>

      </li>
      <li className="treasures">

        <NavLink to="treasures">
          <span className="material-symbols-outlined">
            arrow_circle_right
          </span>
          <div>
            <h2>Treasures</h2>
            <h3>4</h3>
          </div>
        </NavLink>

      </li>
    </ul>
  );
}

export default Categories;
