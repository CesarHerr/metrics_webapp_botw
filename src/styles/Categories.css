.categoryList {
  display: grid;
  grid-template-columns: repeat( 2, 1fr);
  height: 90vh;
  background-color: rgba(86, 138, 228, 0.671);
  padding-top: 4.5rem;
}

.creatures {
  grid-area: 1 / 1 / 1 / 3;
  position: relative; 
  background-image: url('../images/creatures.png');
  background-position: center;
  background-repeat: no-repeat;
  background-color: rgba(86, 138, 228, 0.897);
  background-blend-mode:soft-light;
}

.equipment {
      position: relative; 
    background-image: url('../images/equipment.png');
    background-position: center;
    background-repeat: no-repeat;
    background-color: rgba(63, 99, 165, 0.897);
    background-blend-mode:soft-light;  
}

.materials {
  position: relative; 
background-image: url('../images/materials.png');
background-position: center;
background-repeat: no-repeat;
background-color: rgba(78, 120, 204, 0.897);
background-blend-mode:soft-light;  
}

.monsters {
  position: relative; 
  background-image: url('../images/monsters.png');
  background-position: center;
  background-repeat: no-repeat;
  background-color: rgba(78, 120, 204, 0.897);
  background-blend-mode:soft-light;
}

.treasures {
  position: relative; 
  background-image: url('../images/treasure.png');
  background-position: center;
  background-repeat: no-repeat;
  background-color: rgba(63, 99, 165, 0.884);
  background-blend-mode:soft-light;
}
.categoryList li {
  
}

.categoryList li a {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-end;
  height: 200px;
  padding: 10px;
}

.categoryList li a h3 {
  display: flex;
  flex-direction: row-reverse;
}
