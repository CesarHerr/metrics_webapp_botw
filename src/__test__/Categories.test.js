import '@testing-library/jest-dom';
import { screen, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Categories from '../components/Categories';

describe('Categories', () => {
  it('Renders Categories component', () => {
    const nav = render(
      <MemoryRouter>
        <Categories />
      </MemoryRouter>,
    );

    expect(nav.container).toMatchSnapshot();
  });

  it('Renders Navigator component', () => {
    render(
      <MemoryRouter>
        <Categories />
      </MemoryRouter>,
    );

    const element = screen.getByText('Creatures');

    expect(element).toBeInTheDocument();
  });
});
