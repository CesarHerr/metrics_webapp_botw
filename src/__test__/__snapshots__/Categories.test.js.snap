// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Renders Categories component 1`] = `
<div>
  <ul
    class="categoryList"
  >
    <li
      class="creatures"
    >
      <a
        class="test"
        href="/creatures"
      >
        <span
          class="material-symbols-outlined"
        >
          arrow_circle_right
        </span>
        <div>
          <h2>
            Creatures
          </h2>
          <h3>
            82
          </h3>
        </div>
      </a>
    </li>
    <li
      class="equipment"
    >
      <a
        class=""
        href="/equipment"
      >
        <span
          class="material-symbols-outlined"
        >
          arrow_circle_right
        </span>
        <div>
          <h2>
            Equipment
          </h2>
          <h3>
            184
          </h3>
        </div>
      </a>
    </li>
    <li
      class="materials"
    >
      <a
        class=""
        href="/materials"
      >
        <span
          class="material-symbols-outlined"
        >
          arrow_circle_right
        </span>
        <div>
          <h2>
            Materials
          </h2>
          <h3>
            80
          </h3>
        </div>
      </a>
    </li>
    <li
      class="monsters"
    >
      <a
        class=""
        href="/monsters"
      >
        <span
          class="material-symbols-outlined"
        >
          arrow_circle_right
        </span>
        <div>
          <h2>
            Monsters
          </h2>
          <h3>
            80
          </h3>
        </div>
      </a>
    </li>
    <li
      class="treasures"
    >
      <a
        class=""
        href="/treasures"
      >
        <span
          class="material-symbols-outlined"
        >
          arrow_circle_right
        </span>
        <div>
          <h2>
            Treasures
          </h2>
          <h3>
            4
          </h3>
        </div>
      </a>
    </li>
  </ul>
</div>
`;
